presentation/src/main/java/me/androidbox/presentation/agenda/viewmodel/AgendaViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package me.androidbox.presentation.agenda.viewmodel
+
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import me.androidbox.presentation.agenda.screen.AgendaScreenEvent
+import me.androidbox.presentation.agenda.screen.AgendaScreenState
+import javax.inject.Inject
+
+@HiltViewModel
+class AgendaViewModel @Inject constructor() : ViewModel() {
+
+    private val agendaScreenMutableState = MutableStateFlow(AgendaScreenState())
+    val agendaScreenState = agendaScreenMutableState.asStateFlow()
+
+    fun onAgendaScreenEvent(agendaScreenEvent: AgendaScreenEvent) {
+        when(agendaScreenEvent) {
+            is AgendaScreenEvent.OnDateChanged -> {
+                agendaScreenMutableState.value = agendaScreenState.value.copy(
+                    displayMonth = agendaScreenEvent.month
+                )
```

Comment:
In general, I'd always use `_stateFlow.update { it.copy(...) }` to avoid race conditions. This is how it could happen:

*StateFlow holds value A*
- Thread 1 calls `copy()` on StateFlow with value A
- Before Thread 1 assigns its changed copy, Thread 2 updates StateFlow with value B
- Now, Thread 1 assigns its changed copy, which is still based on value A though, but the latest value is already value B. Therefore, value B would never be considered here.

Not always an issue, but I worked on highly stateful apps with frequently changing states, in which this really bit me, since such bugs are super hard to find.
Using `_state.value = _state.value.copy(...)` is fine if you always call this from a single thread (which you do here). But in ViewModels, you're very likely to not do that as soon as you'd change the state inside a `viewModelScope`.

Reply:
@philipplackner Updated to use the `.update { .... }`